src/display.c

```diff
@@ -94,6 +99,106 @@ static void DrawBorder(LCUI_PaintContext paint)
 	Graph_DrawHorizLine(&paint->canvas, color, 1, pos, end_x);
 }
 
+static void SurfaceRecord_DumpRects(SurfaceRecord record,
+				    LinkedList *rects)
+{
+	LCUI_Rect rect;
+	const int width = Surface_GetWidth(record->surface);
+	const int height = Surface_GetHeight(record->surface);
+
+	/* top left */
+	rect.x = 0;
+	rect.y = 0;
+	rect.width = width / 2;
+	rect.height = height / 2;
+	RectList_SplitWith(&record->rects, &rect, rects);
+
+	/* top right */
+	rect.x = width / 2;
+	rect.width = width - rect.x;
+	RectList_SplitWith(&record->rects, &rect, rects);
+
+	/* bottom left */
+	rect.x = 0;
+	rect.y = height / 2;
+	rect.width = width / 2;
+	rect.height = height - rect.y;
+	RectList_SplitWith(&record->rects, &rect, rects);
+
+	/* bottom right */
+	rect.x = width / 2;
+	rect.width = width - rect.x;
+	RectList_SplitWith(&record->rects, &rect, rects);
+
+	RectList_Clear(&record->rects);
+}
+
+static size_t LCUIDisplay_RenderSurface(SurfaceRecord record)
+{
+	size_t count = 0;
+	int i;
+	LCUI_Rect *rect;
+	LCUI_PaintContext paint;
+	LCUI_SysEventRec ev;
+	LinkedList rects;
+	LinkedListNode *node;
+	LCUI_BOOL can_render;
+	LCUI_Rect **rectArray;
+
+	ev.type = LCUI_PAINT;
+	can_render = record->widget && record->surface &&
+		     Surface_IsReady(record->surface);
+	record->rendered = FALSE;
+	LinkedList_Init(&rects);
+	SurfaceRecord_DumpRects(record, &rects);
+
+	rectArray = (LCUI_Rect **)malloc(sizeof(LCUI_Rect*) * rects.length);
+	i = 0;
```

Comment:
The `i` variable is assigned to 0 in the for loop below, so this line can be removed.